lib/sdr_client/redesigned_client/create_resource.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module SdrClient
+  class RedesignedClient
+    # Creates a resource (metadata) in SDR
+    class CreateResource
+      def self.run(...)
+        new(...).run
+      end
+
+      # @param [Boolean] accession should the accessionWF be started
+      # @param [Boolean] assign_doi should a DOI be assigned to this item
+      # @param [Cocina::Models::RequestDRO, Cocina::Models::RequestCollection] metadata
+      # @param [Hash<Symbol,String>] the result of the metadata call
+      # @param [String] priority what processing priority should be used
+      #                          either 'low' or 'default'
+      def initialize(accession:, metadata:, assign_doi: false, priority: nil)
+        @accession = accession
+        @priority = priority
+        @assign_doi = assign_doi
+        @metadata = metadata
+      end
+
+      # @param [Hash<Symbol,String>] the result of the metadata call
+      # @return [String] job id for the background job result
+      def run
+        response = metadata_request
+        UnexpectedResponse.call(response) unless response.status == 201
+
+        logger.info("Response from server: #{response.body}")
+
+        JSON.parse(response.body)['jobId']
+      end
+
+      private
+
+      attr_reader :metadata, :priority
+
+      def logger
+        SdrClient::RedesignedClient.config.logger
+      end
+
+      def client
+        SdrClient::RedesignedClient.instance
+      end
```

Comment:
Get this stuff from central config (the value of using a singleton) so we don't have to pass it around everywhere.